zipline/data/_adjustments.pyx

```diff
@@ -234,64 +291,50 @@ cpdef load_adjustments_from_sqlite(object adjustments_db,  # sqlite3.Connection
             asset_ixs[sid] = assets.get_loc(sid)
         asset_ix = asset_ixs[sid]
 
-        price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
-        for i, column in enumerate(columns):
-            col_adjustments = results[i]
-            if column != 'volume':
-                try:
-                    col_adjustments[date_loc].append(price_adj)
-                except KeyError:
-                    col_adjustments[date_loc] = [price_adj]
-            else:
-                volume_adj = Float64Multiply(
-                    0, date_loc, asset_ix, asset_ix, 1.0 / ratio
-                )
-                try:
-                    col_adjustments[date_loc].append(volume_adj)
-                except KeyError:
-                    col_adjustments[date_loc] = [volume_adj]
-
-    # mergers affect prices only
-    for sid, ratio, eff_date in mergers:
-        if eff_date < start_date:
-            continue
+        if should_include_price_adjustments:
+            price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
+            price_adjustments.setdefault(date_loc, []).append(price_adj)
 
-        date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
+        if should_include_volume_adjustments:
+            volume_adj = Float64Multiply(
+                0, date_loc, asset_ix, asset_ix, 1.0 / ratio
+            )
+            volume_adjustments.setdefault(date_loc, []).append(volume_adj)
 
-        if not PyDict_Contains(asset_ixs, sid):
-            asset_ixs[sid] = assets.get_loc(sid)
-        asset_ix = asset_ixs[sid]
+    # mergers and dividends affect prices only
+    if should_include_price_adjustments:
+        for sid, ratio, eff_date in mergers:
+            if eff_date < start_date:
+                continue
 
-        adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
-        for i, column in enumerate(columns):
-            col_adjustments = results[i]
-            if column != 'volume':
-                try:
-                    col_adjustments[date_loc].append(adj)
-                except KeyError:
-                    col_adjustments[date_loc] = [adj]
-
-    # dividends affect prices only
-    for sid, ratio, eff_date in dividends:
-        if eff_date < start_date:
-            continue
+            date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
 
-        date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
+            if not PyDict_Contains(asset_ixs, sid):
+                asset_ixs[sid] = assets.get_loc(sid)
+            asset_ix = asset_ixs[sid]
 
-        if not PyDict_Contains(asset_ixs, sid):
-            asset_ixs[sid] = assets.get_loc(sid)
-        asset_ix = asset_ixs[sid]
+            price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
+            price_adjustments.setdefault(date_loc, []).append(price_adj)
+
+        for sid, ratio, eff_date in dividends:
```

Comment:
should we just chain the two sets together with `itertools.chain` to have the same loop body for both sets?